bloock-core/src/record/builder/mod.rs

```diff
@@ -104,9 +108,10 @@ impl Builder {
             let payload = self.document.get_payload();
 
             let encryption = encrypter
-                .encrypt(payload)
+                .encrypt(payload, &[/* TODO */], self.document.headers.ty.clone())
```

Comment:
AES allows you to pass in "associated data" when encrypting, which has to be passed in also when decrypting. If it's different, the decryption will fail. Usually people use things like the user id. In our case I don't see anything we can use so I left it blank for the moment. Maybe we can add something to the document header like a document id or name and use that as associated data? It can be left blank, but having it set adds an extra layer of security.